### ISSUE TITLE
Cannot view any long plaintext guide - cuts out

### ISSUE THREAD

**Issue by ShadowMaker** [hulkrelax/deckfaqs#21]
Any long plaintext guide does not load in full for me, it just cuts on some random place somewhere in the middle, like 50-100 screen pages down. For example you can try this faq https://gamefaqs.gamespot.com/pc/552350-icewind-dale-ii/faqs/68029 and try to reach the end of it (** END OF FILE ** wording) on DeckFAQs.

**Comment by hulkrelax** [comment]
Yeah I think this is similar to other issues with long guides causing issues. I've been super busy with professional work and then on weekends I wanna play video games 😀. I'll try to investigate this stuff this weekend. Can't promise I'll have a solution but at least maybe have a better idea on what's going on.

**Comment by hulkrelax** [comment]
I just looked at this guide and was able to scroll all the way to END OF FILE.

**Comment by ShadowMaker** [comment]
> I just looked at this guide and was able to scroll all the way to END OF FILE.

Thanks for checking. Well, I certainly can not, I can attach you a screenshot if this helps, but I doubt will be much sense in it.

Maybe it has something to do with connection timeout or like that? I have a mobile 3g connection, so can be slow to get, maybe you can try something like your mobile connection too?

**Comment by hulkrelax** [comment]
yeah I mean if you are on a 3g connection that is very different from my setup so not entirely straight forward to reproduce. I don't think even with my phone I get 3G since I'm in a 5G area. It likely is a timeout of some kind. There is a 10 second timeout but it seems like you are able to see at least part of the guide? If you see part of the guide then I think I might know what's going on. GameFAQs has a lot of ads so the code checks the page until the guide has been loaded to prevent waiting for all the ads to load. It's possible that on a slow connection that the guide only partially loads and then as a result only part of the guide shows up. It's a difficult one to address because the alternative is loading guides will get even slower if you have to wait for the entire page to load (because of all of the ads). I'll have to do some testing but I'm not sure how many users are using their Steam Deck on 3G connections.

**Comment by ShadowMaker** [comment]
Yes, I see part of the guide which can cut out in different places. Regarding your opinion about 3G - actually I think a lot of people use it on 3G connection, as people travel here and there, Deck is a portable platform and not everywhere you will get a decent coverage. I see that there is already an issue on this github (albeit called Feature request about refresh) which also tells about guide cut outs. So I think having a non-5G ideal connection is not a rare case at all. Maybe you can introduce a setting for mobile connection to wait for all page to load. I actually not really sure why you wait for those ads for plain-text guides, as you can just load the html source page, get div "faqtext" from it and show to user, probably will be 100% fool-proof, but you know better, how all this works. Thanks :)

**Comment by hulkrelax** [comment]
Like I said, I **don't** wait for the ads. I'm specifically waiting until I can grab the guide text as quickly as possible so end users aren't waiting a long time. I can look into adding an option to wait for the full page to load before grabbing the guide text. Maybe that will help.

**Comment by hulkrelax** [comment]
In regards to how I load the page. I'm loading it in the background in a BrowserView so that the page request appears legitimate. Earlier in development, I was getting banned from GameFAQs for just loading the html. So if anything I'm help everyone out so they don't get banned.

**Comment by jlar0che** [comment]
I just installed DeckFaqs on my Steam Deck and have the same exact problem. Here is the Walkthrough that is not fully loading: https://gamefaqs.gamespot.com/ps/196754-beyond-the-beyond/faqs/3710
